File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/game/BoardRenderer.java
package com.nopalsoft.sokoban.game;

import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.maps.tiled.TiledMapTileLayer;
import com.badlogic.gdx.maps.tiled.renderers.OrthogonalTiledMapRenderer;
import com.nopalsoft.sokoban.Assets;
import com.nopalsoft.sokoban.screens.Screens;

public class BoardRenderer {

    SpriteBatch batch;
    OrthogonalTiledMapRenderer renderer;
    TiledMapTileLayer tileLayer;
    OrthographicCamera camera;

    public BoardRenderer(SpriteBatch batch) {
        this.batch = batch;
        camera = new OrthographicCamera(Screens.SCREEN_WIDTH, Screens.SCREEN_HEIGHT);
        camera.position.set(Screens.SCREEN_WIDTH / 2f, Screens.SCREEN_HEIGHT / 2f, 0);
        renderer = new OrthogonalTiledMapRenderer(Assets.map, GameBoard.UNIT_SCALE);
        tileLayer = (TiledMapTileLayer) renderer.getMap().getLayers().get("StaticMap");
    }

    public void render() {
        camera.update();
        renderer.setView(camera);
        renderer.getBatch().begin();
        renderer.renderTileLayer(tileLayer);
        renderer.getBatch().end();
    }
}

File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/game/BoardRenderer.kt
package com.nopalsoft.sokoban.game

import com.badlogic.gdx.graphics.OrthographicCamera
import com.badlogic.gdx.maps.tiled.TiledMapTileLayer
import com.badlogic.gdx.maps.tiled.renderers.OrthogonalTiledMapRenderer
import com.nopalsoft.sokoban.Assets
import com.nopalsoft.sokoban.screens.BaseScreen

class BoardRenderer {

    private val camera = OrthographicCamera(BaseScreen.SCREEN_WIDTH, BaseScreen.SCREEN_HEIGHT)
    private val tiledRenderer = OrthogonalTiledMapRenderer(Assets.map, Board.UNIT_SCALE)
    private val mapStaticLayer = tiledRenderer.map.layers["StaticMap"] as TiledMapTileLayer

    init {
        camera.position.set(BaseScreen.SCREEN_WIDTH / 2F, BaseScreen.SCREEN_HEIGHT / 2F,0F)
    }

     fun render(){
        camera.update()
        tiledRenderer.setView(camera)
        tiledRenderer.batch.begin()
        tiledRenderer.renderTileLayer(mapStaticLayer)
        tiledRenderer.batch.end()
    }
}